example/README.md

```diff
@@ -27,6 +27,200 @@ DEPLOYMENT=cumulus-from-source jasmine spec/ingestGranule/IngestGranuleSuccessSp
+using [Terraform](https://terraform.io).
+
+If you are using a Mac and [Homebrew](https://brew.sh), installing Terraform is
+as simple as:
+
+```shell
+$ brew update
+$ brew install terraform
+```
+
+For other cases,
+[installation instructions](https://learn.hashicorp.com/terraform/getting-started/install.html)
+are available.
+
+Verify that the version of Terraform installed is at least v0.12.0.
+
+```shell
+$ terraform --version
+Terraform v0.12.2
+```
+
+The state of the Terraform deployment is stored in S3. In the following
+examples, it will be assumed that state is being stored in a bucket called
+`my-tf-state`. You can also use an existing bucket, if desired.
+
+Create the state bucket:
+
+```shell
+$ aws s3api create-bucket --bucket my-tf-state
+```
+
+In order to help prevent loss of state information, it is recommended that
+versioning be enabled on the state bucket:
+
+```shell
+$ aws s3api put-bucket-versioning \
+    --bucket my-tf-state \
+    --versioning-configuration Status=Enabled
+```
+
+Terraform uses a lock stored in DynamoDB in order to prevent multiple
+simultaneous updates. In the following examples, that table will be called
+`my-tf-locks`.
+
+Create the locks table:
+
+⚠️ **Note:** The `--billing-mode` option was recently added to the AWS CLI. You
+may need to upgrade your version of the AWS CLI if you get an error about
+provisioned throughput when creating the table.
+
+```shell
+$ aws dynamodb create-table \
+    --table-name my-tf-locks \
+    --attribute-definitions AttributeName=LockID,AttributeType=S \
+    --key-schema AttributeName=LockID,KeyType=HASH \
+    --billing-mode PAY_PER_REQUEST
+```
+
+The Thin Egress App uses a `bucket_map.yaml` file to determine what buckets to
+serve. Documentation of the file format is available [here](https://github.com/asfadmin/thin-egress-app#bucket-map).
+For Cumulus integration tests, a simple config could look like this:
+
+**bucket_map.yaml**
+```yaml
+MAP:
+  my-protected: my-protected
+
+PUBLIC_BUCKETS:
+  - my-public
+```
+
+Upload `bucket_map.yaml` to your stack's internal bucket:
+
+```shell
+$ aws s3 cp ./bucket_map.yaml s3://my-internal/bucket_map.yaml
+```
+
+The Thin Egress App stores its Earthdata Login credentials in AWS Secrets
+Manager. These are the `EARTHDATA_CLIENT_ID` and `EARTHDATA_CLIENT_PASSWORD`
+values from your `example/app/.env` file. The client id and password need to be
+`base64` encoded. You can get the value by running:
+
+```shell
+$ echo -n 'user:pass' | base64
+```
+
+⚠️ **Note:** All `aws` and `terraform` commands for the rest of this deployment
+_must_ be run using `NGAPShNonProd` credentials.
+
+Store the secrets to Secrets Manager, substituting the appropriate values for
+`UrsId` and `UrsAuth`.
+
+```shell
+aws secretsmanager create-secret \
+    --name my-tea-urs-creds \
+    --secret-string '{"UrsId": "client-id","UrsAuth": "base64-encoded-value"}' \
+    --description 'URS Credentials for My TEA Distribution App'
+```
```

Comment:
Something isn't crystal clear here.  It took me a couple of reads to figure out that the base64-encoded-value should be the output of L116.
Maybe clarifying L112 with the `The client id and password need to be joined and 
base64 encoded. before passed as the UrsAuth value.` . Or something like that.  And If I still have the wrong idea, of what has to happen, it also says it's not clear.

Reply:
Let me know if this is better: https://github.com/nasa/cumulus/pull/1061/files#diff-20bfc3ff6a860483887b93bf9cf0d135R110

Reply:
1d44138 is 💯   Thanks.